rotkehlchen/chain/ethereum/modules/makerdao/decoder.py

```diff
@@ -436,6 +443,67 @@ def decode_cdp_manager_events(self, context: DecoderContext) -> DecodingOutput:
+            to_address = owner_address
+            from_address: Optional[ChecksumEvmAddress] = self.sdai.evm_address
+        elif context.tx_log.topics[0] == SDAI_REDEEM:  # SDAI -> DAI: owner deposits sdai
+            owner_address = self._get_address_or_proxy(hex_or_bytes_to_address(context.tx_log.topics[3]))  # noqa: E501
+            from_address = owner_address
+            to_address = self.sdai.evm_address
+        else:
+            return DEFAULT_DECODING_OUTPUT
+
+        if from_address is None or to_address is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(context.tx_log.data[32:64]),
+            asset=self.sdai,
+        )
+
+        if amount == ZERO:
+            return DEFAULT_DECODING_OUTPUT
+
+        if owner_address == from_address:
+            event_type = HistoryEventType.DEPOSIT
+            event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+            notes = f'Return {amount} sDAI to sDAI pot'
+        else:
+            event_type = HistoryEventType.WITHDRAWAL
+            event_subtype = HistoryEventSubType.REMOVE_ASSET
+            notes = f'Withdraw {amount} sDAI from sDAI pot'
+
+        transfer = self.base.make_event_from_transaction(
+            transaction=context.transaction,
+            tx_log=context.tx_log,
+            event_type=event_type,
+            event_subtype=event_subtype,
+            asset=A_SDAI,
+            balance=Balance(amount=amount),
+            location_label=owner_address,
+            notes=notes,
+            address=self.sdai.evm_address,
+            counterparty=CPT_SDAI,
+        )
+
+        for event in context.decoded_events:
+            if event.event_subtype == HistoryEventSubType.NONE:
+                incoming_event = event.event_type == HistoryEventType.RECEIVE
+                event.event_type = HistoryEventType.WITHDRAWAL if incoming_event else HistoryEventType.DEPOSIT  # noqa: E501
+                event.event_subtype = HistoryEventSubType.REMOVE_ASSET if incoming_event else HistoryEventSubType.DEPOSIT_ASSET  # noqa: E501
+                verb, preposition = ('Withdraw', 'from') if incoming_event else ('Deposit', 'to')
+                event.notes = f'{verb} {event.balance.amount} {self.dai.symbol} {preposition} sDAI pot'  # noqa: E501
+                event.counterparty = CPT_SDAI
+
+                deposit_event, receive_event = (transfer, event) if incoming_event else (event, transfer)  # noqa: E501
+                maybe_reshuffle_events(
+                    ordered_events=[deposit_event, receive_event],
+                    events_list=context.decoded_events,
+                )
+        return DecodingOutput(event=transfer, action_items=[])
+
     def decode_saidai_migration(self, context: DecoderContext) -> DecodingOutput:
         if context.tx_log.topics[0] == ERC20_OR_ERC721_TRANSFER:
             to_address = hex_or_bytes_to_address(context.tx_log.topics[2])
```

Comment:
I see that the `possible_events` method is missing from here. You need to implement it and list all possible even ttypes otherwise the frontend will not render them correctly.

Check how your PR renders for example:


![2023-09-29_15-33](https://github.com/rotki/rotki/assets/1658405/d86b24db-4fd3-4ec0-b2d7-2c9fb72fdaa2)


Notice the question marks